target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+            from_slice,
+            v1::{
+                MerklePriceUpdate,
+                WormholeMessage,
+                WormholePayload,
+            },
+        },
+        ACCUMULATOR_EMITTER_ADDRESS,
+    },
+    serde::Deserialize,
+    sha3::Digest,
+    state::AnchorVaa,
+    std::io::Write,
+    wormhole_anchor_sdk::wormhole::{
+        SignatureSetData,
+        SEED_PREFIX_POSTED_VAA,
+    },
+};
+
+declare_id!("DvPfMBZJJwKgJsv2WJA8bFwUMn8nFd5Xpioc6foC3rse");
+pub const POST_VAA: u8 = 2;
+
+#[program]
+pub mod pyth_solana_receiver {
+    use super::*;
+
+    /// Verify the updates using the posted_vaa account
+    ///  * `vaa_hash` hash post of the post_vaa data to derive the address of the post_vaa account
+    ///  * `price_updates` Vec of bytes for the updates to verify and post on-chain
+    #[allow(unused_variables)]
+    pub fn post_updates(ctx: Context<PostUpdates>, price_updates: Vec<Vec<u8>>) -> Result<()> {
+        let vaa = &ctx.accounts.posted_vaa;
+        let signature_set = &ctx.accounts.signature_set;
+        require_keys_eq!(
+            vaa.meta.signature_set,
+            signature_set.key(),
+            ReceiverError::InvalidSignatureSet
+        );
+
+        let (expected_vaa_pubkey, _) = Pubkey::find_program_address(
+            &[SEED_PREFIX_POSTED_VAA, &signature_set.hash],
+            //TODO: expected program owner of the posted_vaa account should come from config account that can only be modified by governance
+            &wormhole_anchor_sdk::wormhole::program::id(),
+        );
+        require_keys_eq!(
+            vaa.key(),
+            expected_vaa_pubkey,
+            ReceiverError::InvalidVaaAccountKey
+        );
+
+        // TODO: expected emitter_chain should come from config account that can only be modified by governance
+        require_eq!(
+            vaa.emitter_chain(),
+            <wormhole::Chain as Into<u16>>::into(wormhole::Chain::Pythnet),
+            ReceiverError::InvalidEmitterChain
+        );
+
+        require_keys_eq!(
+            Pubkey::new_from_array(*vaa.emitter_address()),
+            // TODO: expected emitter_address should come from config account that can only be modified by governance
+            Pubkey::new_from_array(ACCUMULATOR_EMITTER_ADDRESS),
+            ReceiverError::InvalidEmitterAddress
+        );
+
+        let wh_message = WormholeMessage::try_from_bytes(vaa.payload.as_slice())
+            .map_err(|_| ReceiverError::InvalidWormholeMessage)?;
+        msg!("constructed wh_message {:?}", wh_message);
+        let root: MerkleRoot<Keccak160> = MerkleRoot::new(match wh_message.payload {
+            WormholePayload::Merkle(merkle_root) => merkle_root.root,
+        });
+
+        let mut count_updates = 0;
+
+        let price_updates_len = price_updates.len();
+        for price_update in price_updates {
+            let merkle_price_update =
+                from_slice::<byteorder::BE, MerklePriceUpdate>(price_update.as_slice())
```

Comment:
wait we already have a struct for these updates. Why not pass `Vec<MerklePriceUpdate>` as the argument to this instruction then and then implement whatever trait anchor needs to get the serialization right?

Reply:
`MerklePriceUpdate` comes from an external crate and the `Keccak160` for the proof doesn't implement `BorshSerialize` or `BorshDeserialize`.

 I will look into making a separate PR and add a todo for this so that we can pass in `Vec<MerklePriceUpdate>` directly like you suggested since this PR is already starting to get a bit large.